alicloud/resource_alicloud_ots_table.go

```diff
@@ -0,0 +1,180 @@
+package alicloud
+
+import (
+	"fmt"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/aliyun/aliyun-tablestore-go-sdk/tablestore"
+)
+
+func resourceAlicloudOtsTable() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAliyunOtsTableCreate,
+		Read: resourceAliyunOtsTableRead,
+		Update: resourceAliyunOtsTableUpdate,
+		Delete: resourceAliyunOtsTableDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"table_name": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"primary_key": {
+				Type:     schema.TypeList,
+				Required: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"name": &schema.Schema{
+							Type:     schema.TypeString,
+							Required: true,
+						},
+						"type": &schema.Schema{
+							Type:     schema.TypeString,
+							Required: true,
+							ValidateFunc: validatePrimaryTypeKey,
+						},
+					},
+				},
+				MaxItems: 4,
+			},
+			"table_option": {
+				Type:     schema.TypeSet,
+				Required: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"time_to_live": {
+							Type:     schema.TypeInt,
+							Optional: true,
+							Default: -1,
+						},
+						"max_version": {
+							Type:     schema.TypeInt,
+							Optional: true,
+							Default: 1,
+						},
+					},
+				},
+				MaxItems: 1,
+			},
+		},
+	}
+}
+
+func resourceAliyunOtsTableCreate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*AliyunClient).otsconn
+
+	tableMeta := new(tablestore.TableMeta)
+	table_name := d.Get("table_name").(string)
+	tableMeta.TableName = table_name
+
+	for _, primaryKey :=range d.Get("primary_key").([]interface{}){
+		pk := primaryKey.(map[string]interface{})
+		pkValue := getPrimaryKeyType(pk["type"].(string))
+		tableMeta.AddPrimaryKeyColumn(pk["name"].(string), pkValue)
+
+	}
+	tableOption := new(tablestore.TableOption)
+	to := d.Get("table_option").(*schema.Set).List()
+	w := to[0].(map[string]interface{})
```

Comment:
Whether to need to ensure param 'to' is nil or length is 0. I think, if table_option maxItem is 1, you can make its parameter separate from table_option.